js/packages/wallet-adapter-mobile/src/adapter.ts

```diff
@@ -74,12 +77,9 @@ export class SolanaMobileWalletAdapter extends BaseMessageSignerWalletAdapter {
     }
 
     get publicKey(): PublicKey | null {
-        if (this._publicKey == null && this._authorizationResult != null) {
+        if (this._publicKey == null && this._selectedAddress != null) {
             try {
-                this._publicKey = getPublicKeyFromAddress(
-                    // TODO(#44): support multiple addresses
-                    this._authorizationResult.addresses[0],
-                );
+                this._publicKey = getPublicKeyFromAddress(this._selectedAddress);
```

Comment:
I might be missing something here, but how can you reliably get a public key from an address? Addresses (if not on Solana) may not be reversible to pubkeys.

Reply:
I guess I presumed that the Solana `wallet-adapter` would only ever work with Solana wallets. We’d have to change the return type of `useWallet`, which right now only has the concept of `publicKey`.

Reply:
Ah, that's a fair assumption then! I think I was not seeing that the context of this code is `wallet-adapter-mobile` and not `mobile-wallet-adapter`